calm/draft/2024-10/meta/flow.json

```diff
@@ -0,0 +1,82 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://calm.finos.org/draft/2024-10/meta/flow.json",
+  "title": "Business Flow Model",
+  "description": "Defines business flows that relate to technical architectures, allowing mapping of flows to technical components and attaching control requirements.",
+  "type": "object",
+  "properties": {
+    "unique-id": {
+      "type": "string",
+      "description": "Unique identifier for the flow"
+    },
+    "name": {
+      "type": "string",
+      "description": "Descriptive name for the business flow"
+    },
+    "description": {
+      "type": "string",
+      "description": "Detailed description of the flow's purpose"
+    },
+    "requirement-url": {
+      "type": "string",
+      "description": "Link to a detailed requirement document"
+    },
+    "transitions": {
+      "type": "array",
+      "items": {
+        "$ref": "#/defs/transition"
+      }
+    },
+    "controls": {
+      "$ref": "https://calm.finos.org/draft/2024-10/meta/control.json#/defs/controls"
+    },
+    "metadata": {
+      "$ref": "#/defs/metadata"
+    }
+  },
+  "required": [
+    "unique-id",
+    "name",
+    "description",
+    "transitions"
+  ],
+  "defs": {
+    "transition": {
+      "type": "object",
+      "properties": {
+        "relationship-unique-id": {
+          "type": "string",
+          "description": "Unique identifier for the relationship in the architecture"
+        },
+        "sequence-number": {
+          "type": "integer",
+          "description": "Indicates the sequence of the relationship in the flow"
+        },
+        "summary": {
+          "type": "string",
+          "description": "Functional summary of what is happening in the transition"
+        },
+        "direction": {
```

Comment:
Just this first version makes me realise how it important to define documentation around relationship type definitions and also the important of having the data assets will link to relationships. 

Taking the traderx flow, there are interesting scenarios like 

1. Positions/Trades being published to GUI at same time off the back of trade change.
2. Bidirectional data communication over the same interface. Doesn't feel right to define a distinct relationship for the other way for Rest Sync calls or open websocket communication for certain flows.

Reply:
A request-response interaction should be only a single relationship, but I'm not sure how we could then describe the timing of the response (as per a normal UML sequence diagram).

Reply:
Two transitions tagged to the same relationship?

Reply:
That's what this does

```json
   {
       "relationship-unique-id": "web-gui-process-uses-accounts-service",
       "sequence-number": 3,
       "summary": "Returns Account Create/Update Response Status",
       "direction": "destination-to-source"
     }
```

We're using the direction field to enable us to tell the sequence diagram to reverse the relationship direction.

Reply:
Right, but UI sends trade booking to back-end (transition 1) which does a number of things (transitions 2-7, say) and then returns a response to the UI - is that response transition 8 but to the same relationship as transition 1?

Reply:
Yes, I think that's right.